workflow/controller/operator.go

```diff
@@ -646,10 +646,19 @@ func (woc *wfOperationCtx) reapplyUpdate(wfClient v1alpha1.WorkflowInterface, no
 		if err != nil {
 			return nil, err
 		}
+		if currWf.Status.Fulfilled() {
```

Comment:
something about the indexers (no, I'm not sure what) means we are more likely to operate on the list version of the workflow, and potentially error a previously succeeded workflow - this prevents that